### PR TITLE
fix: parsing of --amd flag

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
fix

**Did you add tests for your changes?**
yes
**If relevant, did you update the documentation?**
no
**Summary**
`amd` can accept only `false` as boolean. Hence only `--no-amd` is valid.

### Before
Screenshot at 2020-11-05 08-38-11

### After
Screenshot at 2020-11-05 08-36-39

**Does this PR introduce a breaking change?**
No

**Other information**
Open to suggestions 😄

### CODE REVIEW

File: packages/webpack-cli/lib/groups/basicResolver.js [hunk 1]
```diff
@@ -30,6 +30,9 @@ function resolveArgs(args) {
         if (arg === 'watch') {
             finalOptions.options.watch = true;
         }
+        if (arg === 'amd') {
+            finalOptions.options.amd = args[arg];
+        }
```
Comment (snitin315) [inline packages/webpack-cli/lib/groups/basicResolver.js:35]: I think here we can tell the user to use `--no-amd` instead. Any thoughts?

Comment (evilebottnawi) [reply packages/webpack-cli/lib/groups/basicResolver.js:35]: Yes, I think we should avoid it in our code, what is webpack.cli return on amd?

Comment (evilebottnawi) [reply packages/webpack-cli/lib/groups/basicResolver.js:35]: If only `false` we should support on `--no-amd`

Comment (snitin315) [reply packages/webpack-cli/lib/groups/basicResolver.js:35]: yes only `amd: false` is allowed.

Comment (evilebottnawi) [reply packages/webpack-cli/lib/groups/basicResolver.js:35]: So let's support only `--no-amd`, do we need fix something on webpack side?

Comment (snitin315) [reply packages/webpack-cli/lib/groups/basicResolver.js:35]: Webpack doesn't export any negative flags. They are added by CLI only. I think we can do something like - 

```js
if (arg === 'amd') {
       if (args[arg] === true) {
           logger.error('configuration.amd can accept only false as boolean. Only --no-amd is supported.');
            process.exit(2);
        }
          finalOptions.options.amd = args[arg];
   }
```

Comment (evilebottnawi) [reply packages/webpack-cli/lib/groups/basicResolver.js:35]: We should avoid put logic for special values of flags inside webpack-cli, we should fix it on webpack side

Comment (snitin315) [reply packages/webpack-cli/lib/groups/basicResolver.js:35]: /cc @sokra

**Comment by alexander-akait** [comment]
Fixed in master, webpack return `problems`, so we output error message when value is invalid, anyway you can help with https://github.com/webpack/webpack/issues/12113 for better DX

**Pull request closed without merging**